backend/src/middlewares/set-trace-id/set-trace-id.middleware.ts

```diff
@@ -1,10 +1,18 @@
 import { RequestHandler } from 'express';
-import { appAsyncStorage } from '~/services/services';
+import { appAsyncStorage, token } from '~/services/services';
 import { getRandomId } from '~/helpers/helpers';
 import { AppAsyncStorageKey } from '~/common/enums/enums';
+import { TokenPayload } from '~/common/types/types';
 
-const setTraceId: RequestHandler = async (_req, _res, next) => {
-  const traceId = getRandomId();
+const setTraceId: RequestHandler = async (req, _res, next) => {
+  const authorizationHeader = req.get('Authorization');
+  const bearerToken = authorizationHeader?.split(' ')[1];
```

Comment:
what does 1 mean?

we already have this functionality in the current-user route

we need reuse it